### PR TITLE
Add kr6 r900-2 support

### PR DESCRIPTION
**Pull request by cjue**
kr6 r900-2 is identical to kr10 r900-2 except for velocity limits, which were updated from the manual
the kr10 r900-2 meshes are reused

### CODE REVIEW

**Comment by simonschmeisser** [comment]
could you please add the opw parameter file similar to this one https://github.com/isys-vision/kuka_experimental/blob/indigo-devel/kuka_kr4_support/config/opw_parameters_kr4r600.yaml

**Comment by cjue** [comment]
done

**Review by simonschmeisser — CHANGES_REQUESTED** [review]
Being a bit pedantic now: could you please also add your model in the package.xml, it contains some hints as to where the data comes from

File: kuka_kr6_support/launch/load_kr6r900_2.launch [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0"?>
+<launch>
+  <param name="robot_description" command="$(find xacro)/xacro.py '$(find kuka_kr6_support)/urdf/kr6r900_2.xacro'"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/launch/load_kr6r900_2.launch]:
```suggestion
  <param name="robot_description" command="$(find xacro)/xacro '$(find kuka_kr6_support)/urdf/kr6r900_2.xacro'"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    </link>
+    <!-- This frame corresponds to the $FLANGE coordinate system in KUKA KRC controllers. -->
+    <link name="${prefix}tool0"/>
+
+    <joint name="${prefix}joint_a1" type="revolute">
+      <origin xyz="0 0 0.400" rpy="0 0 0"/>
+      <parent link="${prefix}base_link"/>
+      <child link="${prefix}link_1"/>
+      <axis xyz="0 0 -1"/>
+      <limit effort="0" lower="${-DEG2RAD * 170}" upper="${DEG2RAD * 170}" velocity="${DEG2RAD * 360}"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <limit effort="0" lower="${radians(-170)}" upper="${radians(170)}" velocity="${radians(360)}"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    </joint>
+    <joint name="${prefix}joint_a2" type="revolute">
+      <origin xyz="0.025 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_1"/>
+      <child link="${prefix}link_2"/>
+      <axis xyz="0 1 0"/>
+      <limit effort="0" lower="${-DEG2RAD * 190}" upper="${DEG2RAD * 45}" velocity="${DEG2RAD * 300}"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <limit effort="0" lower="${radians(-190)}" upper="${radians(45)}" velocity="${radians(300)}"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    </joint>
+    <joint name="${prefix}joint_a3" type="revolute">
+      <origin xyz="0.455 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_2"/>
+      <child link="${prefix}link_3"/>
+      <axis xyz="0 1 0"/>
+      <limit effort="0" lower="${-DEG2RAD * 120}" upper="${DEG2RAD * 156}" velocity="${DEG2RAD * 360}"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <limit effort="0" lower="${radians(-120)}" upper="${radians(156)}" velocity="${radians(360)}"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+    </joint>
+    <joint name="${prefix}joint_a4" type="revolute">
+      <origin xyz="0 0 0.025" rpy="0 0 0"/>
+      <parent link="${prefix}link_3"/>
+      <child link="${prefix}link_4"/>
+      <axis xyz="-1 0 0"/>
+      <limit effort="0" lower="${-DEG2RAD * 185}" upper="${DEG2RAD * 185}" velocity="${DEG2RAD * 450}"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <limit effort="0" lower="${radians(-185)}" upper="${radians(185)}" velocity="${radians(450)}"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    </joint>
+    <joint name="${prefix}joint_a5" type="revolute">
+      <origin xyz="0.420 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_4"/>
+      <child link="${prefix}link_5"/>
+      <axis xyz="0 1 0"/>
+      <limit effort="0" lower="${-DEG2RAD * 120}" upper="${DEG2RAD * 120}" velocity="${DEG2RAD * 450}"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <limit effort="0" lower="${radians(-120)}" upper="${radians(120)}" velocity="${radians(450)}"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+    </joint>
+    <joint name="${prefix}joint_a6" type="revolute">
+      <origin xyz="0.090 0 0" rpy="0 0 0"/>
+      <parent link="${prefix}link_5"/>
+      <child link="${prefix}link_6"/>
+      <axis xyz="-1 0 0"/>
+      <limit effort="0" lower="${-DEG2RAD * 350}" upper="${DEG2RAD * 350}" velocity="${DEG2RAD * 540}"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <limit effort="0" lower="${radians(-350)}" upper="${radians(350)}" velocity="${radians(540)}"/>
```

File: kuka_kr6_support/urdf/kr6r900_2_macro.xacro [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+    </joint>
+    <joint name="${prefix}joint_a6-tool0" type="fixed">
+      <parent link="${prefix}link_6"/>
+      <child link="${prefix}tool0"/>
+      <origin xyz="0 0 0" rpy="0 ${DEG2RAD * 90} 0"/>
```
Comment (simonschmeisser) [inline kuka_kr6_support/urdf/kr6r900_2_macro.xacro]:
```suggestion
      <origin xyz="0 0 0" rpy="0 ${radians(90)} 0"/>
```

**Review by simonschmeisser — CHANGES_REQUESTED** [review]
I added some suggestions that I'll commit now

**Pull request merged**